include/boost/geometry/algorithms/detail/direction_code.hpp

```diff
@@ -87,42 +87,59 @@ struct direction_code_impl<cartesian_tag>
 template <>
 struct direction_code_impl<spherical_equatorial_tag>
 {
-    template <typename Point1, typename Point2>
-    static inline int apply(Point1 const& segment_a, Point1 const& segment_b,
+    template <typename PointSegmentA, typename PointSegmentB, typename Point2>
+    static inline int apply(PointSegmentA const& segment_a, PointSegmentB const& segment_b,
                             Point2 const& p)
     {
-        typedef typename coordinate_type<Point1>::type coord1_t;
-        typedef typename coordinate_type<Point2>::type coord2_t;
-        typedef typename cs_angular_units<Point1>::type units_t;
-        typedef typename cs_angular_units<Point2>::type units2_t;
-        BOOST_GEOMETRY_STATIC_ASSERT(
-            (std::is_same<units_t, units2_t>::value),
-            "Not implemented for different units.",
-            units_t, units2_t);
-
-        typedef typename geometry::select_coordinate_type <Point1, Point2>::type calc_t;
-        typedef math::detail::constants_on_spheroid<coord1_t, units_t> constants1;
-        typedef math::detail::constants_on_spheroid<coord2_t, units_t> constants2;
-        static coord1_t const pi_half1 = constants1::max_latitude();
-        static coord2_t const pi_half2 = constants2::max_latitude();
+        {
+            using units_sa_t =  typename cs_angular_units<PointSegmentA>::type;
+            using units_sb_t =  typename cs_angular_units<PointSegmentB>::type;
+            using units_p_t = typename cs_angular_units<Point2>::type;
+            BOOST_GEOMETRY_STATIC_ASSERT(
+                (std::is_same<units_sa_t, units_sb_t>::value),
+                "Not implemented for different units.",
+                units_sa_t, units_sb_t);
+            BOOST_GEOMETRY_STATIC_ASSERT(
+                (std::is_same<units_sa_t, units_p_t>::value),
+                "Not implemented for different units.",
+                units_sa_t, units_p_t);
+        }
+
+        using coor_sa_t = typename coordinate_type<PointSegmentA>::type;
+        using coor_sb_t = typename coordinate_type<PointSegmentB>::type;
+        using coor_p_t = typename coordinate_type<Point2>::type;
+
+        // Declare unit type (equal for all types) and calc type (coerced to most precise)
+        using units_t = typename cs_angular_units<Point2>::type;
```

Comment:
This is `units_p_t` above. Should we reuse that type?

Reply:
:heavy_check_mark:

Reply:
Put it back, the one above was in a separate block (specifically for the assertions). I could move that too, but then it's actually clearer like it is now.